### PR TITLE
fix issue with lift_fresh_copy when using export + compile

### PR DESCRIPTION
**Pull request by bdhirsh**
Fixes https://github.com/pytorch/pytorch/issues/105327. The problem is that `lift_fresh_copy()`'s functionalization implementation currently assumes that the input is always not functional. This is apparently too limiting: when you have "user" code like this (which can potentially come from exporting a model and then running compile on the resulting graph):
```
tensor_constant0 = torch.tensor(2)
lift_fresh = torch.ops.aten.lift_fresh_copy.default(tensor_constant0)
```

When we run this through AOTAutograd, the first call (torch.tensor(2)) will **already** be lifted into a functional tensor wrapper - so the `lift_fresh_copy` call doesn't need to do any "lifting" anymore - it just needs to do a clone.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #108570
* __->__ #108243
* #108235
* #108081

### CODE REVIEW

File: aten/src/ATen/FunctionalizeFallbackKernel.cpp [hunk 1]
```diff
@@ -181,14 +181,26 @@ static at::Tensor lift_functionalize(const at::Tensor & self) {
 }
 
 static at::Tensor lift_fresh_functionalize(const at::Tensor & self) {
-  TORCH_INTERNAL_ASSERT(!at::functionalization::impl::isFunctionalTensor(self));
+  // See Note [Exporting and compiling a graph with lift_fresh_copy]
+  if (at::functionalization::impl::isFunctionalTensor(self)) {
+    return self;
```
Comment (albanD) [inline aten/src/ATen/FunctionalizeFallbackKernel.cpp:186]: You CANNOT return self here as you're below autograd.
```suggestion
    return self.view_as(self);
```

Comment (bdhirsh) [reply aten/src/ATen/FunctionalizeFallbackKernel.cpp:186]: And I thought i'd finally drilled that into my head

**Review by albanD — APPROVED** [review]
Sounds fair!

**Pull request closed without merging**